src/io/utilities/readSBML.m

```diff
@@ -636,10 +640,10 @@
     model.lb = fbc_lb;
     model.ub = fbc_ub;
     if noObjective==0; % when there is an objective function
-        indexObj=findRxnIDs(model,fbc_obj);
+        indexObj=findRxnIDs(model,cleanUpFormatting(fbc_obj));
         % indexObj=find(strcmp(fbc_obj,model.rxns))
-        model.c(indexObj)=1;
-        model.osense = - sign(fbc_obj_value);
+        model.c(indexObj) = fbc_obj_value(:);
+        model.osense = fbc_obj_sense;
```

Comment:
@laurentheirendt @tpfau 
Thank you. So can we agree on the changes? If so, are the following what I should do? 
(1) add the test file described in #375, and 
(2) create a new pull request (to the develop branch) cherry-picking the two commits here and the new commit adding the test file, with reference to these two pull requests #375 and #378.

Am I right? Thanks for helping.